### PR TITLE
nfq: Ensure packet release function set 

### PR DESCRIPTION
**Pull request by jlucovsky**
Continuation of #8608 

This PR ensures that early error returns in `NFQCallBack` will have the packet's release function pointer set.

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [5916](https://redmine.openinfosecfoundation.org/issues/5916)

Describe changes:
- Set packet's `ReleasePacket` earlier for early error paths.

Updates:
- Moved setting of `ReleasePacket` to `NFQSetupPkt`

#suricata-verify-pr:
#suricata-verify-repo:
#suricata-verify-branch:
#suricata-update-pr:
#suricata-update-repo:
#suricata-update-branch:
#libhtp-pr:
#libhtp-repo:
#libhtp-branch:

### CODE REVIEW

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW1_stats_chk | | |
| .tcp.rst | 113267 | 145152 | 128.15% |
| | SURI_TLPR1_stats_chk | | |
| .tcp.rst | 6790851 | 6405354 | 94.32% |
| | TREX_GENERIC_stats_chk | | |
| .capture.kernel_drops | 0 | 204 | 0.00 | 

Pipeline 12838

File: src/source-nfq.c [hunk 1]
```diff
@@ -407,6 +408,8 @@ static int NFQSetupPkt (Packet *p, struct nfq_q_handle *qh, void *data)
     char *pktdata;
     struct nfqnl_msg_packet_hdr *ph;
 
+    p->ReleasePacket = NFQReleasePacket;
```
Comment (victorjulien) [inline src/source-nfq.c:411]: in the repeat mode block below we can already set a verdict and return. I think in this case we'd still call NFQReleasePacket, but we probably didn't set p->nfq_v.verdicted, so we'd verdict again?

Comment (jlucovsky) [reply src/source-nfq.c:411]: Should the repeat mode block use `NFQSetVerdict` instead of doing it inline?

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW1_stats_chk | | |
| .tcp.rst | 113181 | 145088 | 128.19% |
| | TREX_GENERIC_stats_chk | | |
| .capture.kernel_drops | 0 | 9 | 0.00 | 

Pipeline 13001

File: src/source-nfq.c (hunk 1 above)
Comment (jlucovsky) [reply src/source-nfq.c]: I've updated this to address the issue you raised.

**Pull request closed without merging**